### ISSUE TITLE
Gnome session

### ISSUE THREAD

**Issue by barnscott** [WhitewaterFoundry/Fedora-Remix-for-WSL#82]
**Is your feature request related to a problem? Please describe.**
I am interested in starting a gnome session via WSL/Fedora-remix, but I am not sure if it is possible. If it is possible, feel free to redirect me to the docs. I'm using WSL2 and FedoraRemix release 32. I understand Gnome would be a resource hog, but I have a fairly powerful PC and am more interested in gaining access to my preferred DE without requiring a VM.

I can currently launch a few GUI apps without issue, like gedit. However, I get the following errors when attempting to run a gnome-session, see below the os-release.

I am using VcXsrv.

$ cat /etc/os-release
```
NAME=Generic
VERSION="32 (Generic)"
ID=generic
ID_LIKE=fedora
VERSION_ID=32
PRETTY_NAME="Generic 32 (Generic)"
ANSI_COLOR="0;34"
LOGO=generic-logo-icon
CPE_NAME="cpe:/o:generic:generic:32"
HOME_URL="http://www.zombo.com/"
SUPPORT_URL="https://en.wikipedia.org/wiki/Help!_(album)"
BUG_REPORT_URL="https://youtu.be/CSemARaqGqE"
REDHAT_BUGZILLA_PRODUCT="Generic"
REDHAT_BUGZILLA_PRODUCT_VERSION=%{bug_version}
REDHAT_SUPPORT_PRODUCT="Generic"
REDHAT_SUPPORT_PRODUCT_VERSION=%{bug_version}
PRIVACY_POLICY_URL="http://nsa.gov"
```

$ gnome-session
```
** (process:1709): WARNING **: 12:48:14.828: Could not make bus activated clients aware of XDG_CURRENT_DESKTOP=GNOME environment variable: Failed to execute child process ?dbus-launch? (No such file or directory)
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
gnome-session-binary[1709]: WARNING: Falling back to non-systemd startup procedure due to error: Failed to execute child process “dbus-launch” (No such file or directory)
gnome-session-binary[1709]: WARNING: Could not make bus activated clients aware of QT_IM_MODULE=ibus environment variable: Failed to execute child process “dbus-launch” (No such file or directory)
gnome-session-binary[1709]: WARNING: Could not make bus activated clients aware of XMODIFIERS=@im=ibus environment variable: Failed to execute child process “dbus-launch” (No such file or directory)
gnome-session-binary[1709]: WARNING: Could not make bus activated clients aware of XDG_MENU_PREFIX=gnome- environment variable: Failed to execute child process “dbus-launch” (No such file or directory)
gnome-session-binary[1709]: WARNING: Using null backend for session tracking
gnome-session-binary[1709]: WARNING: Lost name on bus: org.gnome.SessionManager
Unable to init server: Could not connect: Connection refused

** (gnome-session-failed:1747): WARNING **: 12:48:15.573: Cannot open display:
Terminated
```

$ startx
```
grep: /proc/net/unix: No such file or directory
xauth:  file /home/barn/.serverauth.1748 does not exist

config/udev: failed to bind the udev monitor
[config] failed to pre-init udev

X.Org X Server 1.20.8
X Protocol Version 11, Revision 0
Build Operating System:  5.5.10-200.fc31.x86_64
Current Operating System: Linux PCNAME 4.4.0-19041-Microsoft #1-Microsoft Fri Dec 06 14:06:00 PST 2019 x86_64
Kernel command line: BOOT_IMAGE=/kernel init=/init
Build Date: 30 March 2020  12:00:00AM
Build ID: xorg-x11-server 1.20.8-1.fc32
Current version of pixman: 0.40.0
        Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
Markers: (--) probed, (**) from config file, (==) default setting,
        (++) from command line, (!!) notice, (II) informational,
        (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
(==) Log file: "/var/log/Xorg.0.log", Time: Fri Jun 12 12:49:37 2020
(==) Using system config directory "/usr/share/X11/xorg.conf.d"
(EE)
Fatal server error:
(EE) xf86OpenConsole: Switching VT failed
(EE)
(EE)
Please consult the Fedora Project support
         at http://wiki.x.org
 for help.
(EE) Please also check the log file at "/var/log/Xorg.0.log" for additional information.
(EE)
(EE) Server terminated with error (1). Closing log file.
```

Help appreciated.

**Comment by crramirez** [comment]
We have tried to make run gnome in WSL without success. I can give you instructions to install mate (gnome 2 fork) if you wish.

Regards

**Comment by barnscott** [comment]
Yes please.
And if gnome is not in scope for future release, we can close this "issue".

**Comment by crramirez** [comment]
https://www.whitewaterfoundry.com/blog/2020/4/10/installing-mate-in-fedora-remix-for-wsl

Let's leave it open to keep trying

**Comment by milesbuckton** [comment]
Any news on getting gnome to work for release 33?

**Comment by crramirez** [comment]
Hello @milesbuckton,

Gnome desktop is by far the most difficult desktop to run in WSL. You can easily run MATE and with a bit of work KDE Plasma. But gnome desktop is dependant on systemd and other features. We don't have yet the support for systems. After that, we will try again. 

Meanwhile, if you feel brave enough here is a guide to run Gnome desktop on Ubuntu:

https://www.most-useful.com/ubuntu-20-04-desktop-gui-on-wsl-2-on-surface-pro-4.html#login_with_vcxsrv_in_xdmcp_mode 

I still recommend mate if you want gnome feeling

Regards,
Carlos

**Comment by milesbuckton** [comment]
Hello @crramirez, thank you for your reply. I most certainly agree that it is the most difficult to run in WSL2! I accepted your challenge, and got Gnome to work on Debian, Ubuntu, and Fedora WSL2. 😄 I did come across similar information regarding your link in my previous travels, but I did needed to reference a lot of other pages to come to the complete solution which included network manager, and pulseaudio support. The link that really sealed the deal in assisting me to get it working was: https://github.com/arkane-systems/genie/issues/98#issuecomment-778641443, or similarly https://gist.github.com/gbraad/9b995bfacec3775c297ce89764ba8d6d. 😉 The only real difference was that I installed the "Fedora Workstation" instead of the "Basic Desktop".

![Initial-Setup](https://user-images.githubusercontent.com/37063023/108345785-56a57480-71e7-11eb-8884-c1e4818edcda.png)
![Tour](https://user-images.githubusercontent.com/37063023/108345794-5907ce80-71e7-11eb-93c5-30bf08abd25d.png)
![About](https://user-images.githubusercontent.com/37063023/108345801-5a38fb80-71e7-11eb-8441-fdbb723e7869.png)

**Comment by gbraad** [comment]
@milesbuckton Glad to see this works for you! "Basic Desktop" is my general goto, as people might not prefer using GNOME from WSL2. I personally prefer `i3` myself... and that is not to create a "please, don't touch my computer"-setup ;-)

**Comment by milesbuckton** [comment]
@gbraad, of course...fortunately I have the hardware to run a more bloated Gnome setup. 😄 I did try flatpak, and it was possible to install Discord for example:

![Flatpak_Install](https://user-images.githubusercontent.com/37063023/108467916-9d01de80-728e-11eb-8045-9c508bcdd629.png)
![Flatpak_Installed](https://user-images.githubusercontent.com/37063023/108467919-9ecba200-728e-11eb-87a7-dc60bd305019.png)

**Comment by gbraad** [comment]
@milesbuckton I am actually amazed. I tried LibreOffice this way and I run into issues related to the X-server: "Failed to open display", and for Discord I get issues with "Can't mount tmpfs on /newroot/dev/shm".

**Comment by linghengqian** [comment]
- systemd is fully available in WSL2 and Snap is already available for Ubuntu WSL. Will this issue be made easier? 

- Of course, flatpak is still in a near-unusable state. Like `Can't mount tmpfs on /newroot/dev/shm`.

**Comment by crramirez** [comment]
Yes in fact it should be resolved.

Here are the instructions for SystemD, snap and Gnome. https://www.whitewaterfoundry.com/blog/2022/05/10/fedora-remix-for-wsl-36-released

Fedora Remix supports SystemD in Windows 10 as well.

I haven't had problems with flatpaks but I remember this shm issue. I can check it again but I remember that can be solved mounting it in the right path.

Regards

**Comment by crramirez** [comment]
Tested flatpak with Libre office and it works also
